src/librbd/internal.cc

```diff
@@ -1701,16 +1701,20 @@ int mirror_image_disable_internal(ImageCtx *ictx, bool force,
       uint64_t new_features;
       if (enabled) {
         features &= ~ictx->features;
+	if (features == 0) {
```

Comment:
I think you would need to return -EINVAL if any of the requested feature bits are already enabled.  For example, if object-map is already enabled and the user attempted to enable object-map and fast-diff.